Content/Items/Magnet/Weapons.GrayGoo.cs

```diff
@@ -0,0 +1,435 @@
+using StarlightRiver.Content.Buffs.Summon;
+using StarlightRiver.Content.Dusts;
+using System.Collections.Generic;
+using System.Linq;
+using Terraria.DataStructures;
+using Terraria.ID;
+using static Terraria.ModLoader.ModContent;
+
+namespace StarlightRiver.Content.Items.Magnet
+{
+	public class GrayGooDustData //Has to be a class so we can pass by reference
+	{
+		public int x;
+
+		public int y;
+
+		public Projectile proj;
+
+		public float speed;
+
+		public float lerp;
+		public GrayGooDustData(int x, int y, Projectile proj, float lerp, float speed)
+		{
+			this.x = x;
+			this.y = y;
+			this.proj = proj;
+			this.lerp = lerp;
+			this.speed = speed;
+		}
+	}
+
+	public class GrayGoo : ModItem
+	{
+		public override string Texture => AssetDirectory.MagnetItem + "GrayGoo";
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Gray Goo");
+			Tooltip.SetDefault("Summons a swarm of nanomachines to devour enemies");
+			ItemID.Sets.GamepadWholeScreenUseRange[Item.type] = true; // This lets the Player target anywhere on the whole screen while using a controller.
+			ItemID.Sets.LockOnIgnoresCollision[Item.type] = true;
+		}
+
+		public override void SetDefaults()
+		{
+			Item.damage = 20;
+			Item.mana = 10;
+			Item.width = 32;
+			Item.height = 32;
+			Item.useTime = 36;
+			Item.useAnimation = 36;
+			Item.useStyle = ItemUseStyleID.Swing;
+			Item.value = Item.sellPrice(0, 2, 0, 0);
+			Item.rare = ItemRarityID.Orange;
+			Item.UseSound = SoundID.Item44;
+
+			Item.noMelee = true;
+			Item.DamageType = DamageClass.Summon;
+			Item.buffType = BuffType<GrayGooSummonBuff>();
+			Item.shoot = ProjectileType<GrayGooProj>();
+			Item.knockBack = 0;
+			Item.noUseGraphic = true;
+		}
+
+		public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+		{
+			// This is needed so the buff that keeps your minion alive and allows you to despawn it properly applies
```

Comment:
Are these copy/paste examplemod comments? Do we really need these?

Reply:
fixed